third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -469,6 +496,30 @@ impl ControlFlowGraphCodeGenerator {
                         .collect::<BTreeSet<_>>()
             );
         }
+        // check invariant 3
+        for (block, code) in self.code_blocks.iter() {
+            if block == &self.entry_block || block == &self.exit_block {
+                assert!(code.is_empty())
+            } else {
+                assert!(!code.is_empty(), "block {} is empty", block);
+            }
+        }
+        // check invariant 4
+        assert!(self.succs(self.entry_block).len() == 1);
+        // check invariant 5
+        for (block, succs) in self.successors.iter() {
+            if block != &self.exit_block {
+                assert!(!succs.is_empty(), "block {} has no successors", block);
+            }
+        }
+        // check invariant 6
+        for (block, preds) in self.predecessors.iter() {
+            if *block == self.exit_block || *block == self.get_the_non_trivial_successor(self.entry_block) || preds.is_empty() {
+                continue;
+            }
+            assert!(matches!(self.code_blocks.get(block).expect("code block").first().expect("first instruction"), Bytecode::Label(..)));
+        }
+        // TODO: check other invariants
```

Comment:
There is only one invariant left, why not check that one too, especially as we doing the heavier compare testing process? It will also help make clear what "consistent" precisel means in the invariant description.

Reply:
checked now